src/main/native/cpp/view/Analyzer.cpp

```diff
@@ -231,31 +199,49 @@ void Analyzer::Display() {
         Calculate();
       }
 
-      auto size = ImGui::GetIO().DisplaySize;
-      ImGui::SetNextWindowSize(ImVec2(size.x / 2.5, size.y * 0.9));
+      ImGui::SetCursorPosX(ImGui::GetFontSize() * 15);
+      ImGui::SetNextItemWidth(ImGui::GetFontSize() * 4);
+      if (ImGui::SliderFloat("Test Duration", &m_settings.stepTestDuration,
+                             m_manager->GetMinDuration(),
+                             m_manager->GetMaxDuration(), "%.2f")) {
+        PrepareData();
+        Calculate();
+      }
+
+      // Come back to the starting y pos.
+      ImGui::SetCursorPosY(beginY);
 
-      // Show voltage domain diagnostic plots.
-      if (ImGui::BeginPopupModal("Voltage-Domain Diagnostics")) {
-        m_plot.DisplayVoltageDomainPlots();
-        // Button to close popup.
-        if (ImGui::Button("Close")) {
-          ImGui::CloseCurrentPopup();
-        }
-        ImGui::EndPopup();
+      ShowGain("Ks", &m_ff[0]);
+      ShowGain("Kv", &m_ff[1]);
+      ShowGain("Ka", &m_ff[2]);
+
+      if (m_type == analysis::kElevator) {
+        ShowGain("Kg", &m_ff[3]);
+      } else if (m_type == analysis::kArm) {
+        ShowGain("Kcos", &m_ff[3]);
+      }
+
+      if (m_trackWidth) {
+        ShowGain("Track Width", &*m_trackWidth);
       }
 
-      ImGui::SetNextWindowSize(ImVec2(size.x / 2.5, size.y * 0.9));
+      ShowGain("r-squared", &m_rs);
+
+      ImGui::Separator();
+
+      ImGui::BeginChild("Diagnostic Plots");
+      m_plot.DisplayVoltageDomainPlots();
+      m_plot.DisplayTimeDomainPlots();
 
-      // Show time domain diagnostic plots.
-      if (ImGui::BeginPopupModal("Time-Domain Diagnostics")) {
-        m_plot.DisplayTimeDomainPlots();
-        // Button to close popup.
-        if (ImGui::Button("Close")) {
-          ImGui::CloseCurrentPopup();
+      if (m_dataLoaded) {
+        if (m_graphLoaded) {
```

Comment:
What's the difference between m_dataLoaded and m_graphLoaded? Seems like m_graphLoaded just follows whatever m_dataLoaded does.

The purpose of this logic is to be an edge detector: "if the data wasn't loaded before and it is now".

Reply:
should it be prevDataLoaded?

Reply:
I think so. It makes it clearer what it's for.